src/Swarm/Language/Syntax.hs

```diff
@@ -323,12 +323,15 @@ data Const
     AppF
   | -- God-like sensing operations
 
-    -- | Run a command as if you were another robot.
+    -- | Hypothetically run a command as if you were another robot.
+    --   Make sure this is the LAST command!
     As
+  | -- | Check if the running command has finished.
+    Poll
   deriving (Eq, Ord, Enum, Bounded, Data, Show)
 
 allConst :: [Const]
-allConst = [minBound .. maxBound]
+allConst = [minBound .. As] -- no Poll in code!
```

Comment:
Right, see, this is telling us that we really should make poll a type of `Frame` instead of a `Const`. =)

Reply:
I went full commands instead. :slightly_smiling_face: